include/sparrow/array_data_factory.hpp

```diff
@@ -0,0 +1,347 @@
+// Copyright 2024 Man Group Operations Limited
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or mplied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+#pragma once
+
+#include <algorithm>
+#include <cstddef>
+#include <cstdint>
+#include <functional>
+#include <iterator>
+#include <limits>
+#include <numeric>
+#include <ranges>
+#include <type_traits>
+#include <unordered_map>
+#include <utility>
+
+#include "sparrow/array_data.hpp"
+#include "sparrow/contracts.hpp"
+#include "sparrow/data_traits.hpp"
+#include "sparrow/data_type.hpp"
+#include "sparrow/dictionary_encoded_layout.hpp"
+#include "sparrow/fixed_size_layout.hpp"
+#include "sparrow/memory.hpp"
+#include "sparrow/mp_utils.hpp"
+#include "sparrow/variable_size_binary_layout.hpp"
+
+namespace sparrow
+{
+    template <typename T>
+    array_data for_fixed_size_layout()
```

Comment:
To discuss, the naming of the functions

Reply:
Could you add a description (comments) of the intent of each function? That would help with the naming.

Reply:
Thinking more about this, how about `make_array_data_fixed_layout`?

Reply:
Sounds good, I modified the names